spec/classes/broker_spec.rb

```diff
@@ -2,172 +2,140 @@
-        it { is_expected.to contain_file('/opt/kafka/config/server.properties') }
+      let :common_params do
+        {
+          config: {
+            'zookeeper.connect' => 'localhost:2181'
+          }
+        }
       end
-    end
-
-    describe 'kafka::broker::service' do
-      context 'manage_service false' do
-        let :params do
-          common_params.merge(manage_service: false)
-        end
-
-        it { is_expected.not_to contain_file('/etc/init.d/kafka') }
 
-        it { is_expected.not_to contain_service('kafka') }
+      let :params do
+        common_params
       end
-      context 'defaults' do
-        it { is_expected.to contain_file('/etc/init.d/kafka') }
-
-        context 'limit_nofile set' do
-          let :params do
-            {
-              limit_nofile: '65536'
-            }
-          end
-
-          it { is_expected.to contain_file('/etc/init.d/kafka').with_content %r{ulimit -n 65536$} }
-        end
 
-        context 'limit_core set' do
-          let :params do
-            {
-              limit_core: 'infinity'
-            }
-          end
+      it { is_expected.to contain_class('kafka::broker::install').that_comes_before('Class[kafka::broker::config]') }
+      it { is_expected.to contain_class('kafka::broker::config').that_comes_before('Class[kafka::broker::service]') }
+      it { is_expected.to contain_class('kafka::broker::service').that_comes_before('Class[kafka::broker]') }
+      it { is_expected.to contain_class('kafka::broker') }
 
-          it { is_expected.to contain_file('/etc/init.d/kafka').with_content %r{ulimit -c infinity$} }
+      describe 'kafka::broker::install' do
+        context 'defaults' do
+          it { is_expected.to contain_class('kafka') }
         end
-
-        it { is_expected.to contain_service('kafka') }
-      end
-    end
-  end
-
-  context 'on Centos' do
-    let :facts do
-      {
-        osfamily: 'RedHat',
-        os: { family: 'RedHat' },
-        operatingsystem: 'CentOS',
-        operatingsystemrelease: '7',
-        operatingsystemmajrelease: '7',
-        architecture: 'amd64',
-        path: '/usr/local/sbin',
-        service_provider: 'systemd'
-      }
-    end
-
-    describe 'kafka::broker::install' do
-      context 'defaults' do
-        it { is_expected.to contain_class('kafka') }
       end
-    end
 
-    describe 'kafka::broker::config' do
-      context 'defaults' do
-        it { is_expected.to contain_file('/opt/kafka/config/server.properties') }
+      describe 'kafka::broker::config' do
+        context 'defaults' do
+          it { is_expected.to contain_file('/opt/kafka/config/server.properties') }
+        end
       end
-    end
 
-    describe 'kafka::broker::service' do
-      context 'manage_service false' do
-        let :params do
-          common_params.merge(manage_service: false)
-        end
+      case facts[:service_provider]
+      when 'systemd'
+        describe 'kafka::broker::service' do
```

Comment:
I'd do something like:
```suggestion
        describe 'kafka::broker::service on systemd', if: os_facts[:service_provider] == 'systemd' do
```

Later you can use `unless: ...` as well.

Alternatively you can merge them. For example, with manage_service false, you can expect it not to contain both and have a single test.

Reply:
I just changed it and found an error related to ```limit_nofile``` and ```limit_core``` for non Systemd environments.